sympy/simplify/trigsimp.py

```diff
@@ -426,12 +426,55 @@ def analyse_gens(gens, hints):
 _trigs = (TrigonometricFunction, HyperbolicFunction)
 
 
-def trigsimp(expr, **opts):
+def _trigsimp_inverse(rv):
+
+    funcs = [(sin, asin), (cos, acos), (sec, asec), (csc, acsc),
+             (tan, atan), (cot, acot)]
+
+    def check_args(x, y):
+        try:
+            return x.args[0] == y.args[0]
+        except IndexError:
+            return False
+
+    def f(rv):
+        # for simple functions
+        for func, funcinv in funcs:
+            if isinstance(rv, funcinv) and isinstance(rv.args[0], func):
+                return f(rv.args[0].args[0])
```

Comment:
```python
f = getattr(rv, 'inverse', None)
if f is not None and isinstance(rv.args[0], f()) and isinstance(f(), TrigonometricFunction):
    return f()(rv.args[0].args[0])
... atan2 handling
```

Reply:
Interesting, I don't know how to use this though. 
I ran:
```python
from sympy import sin
from sympy.abc import alpha
rv = sin(alpha)
hasattr(rv, 'inverse')
```
And got `False`

Reply:
`asin` has the inverse, not `sin`.

Reply:
`isinstance(f(), TrigonometricFunction)` gives `False`, while `isinstance(f()(alpha), TrigonometricFunction)` gives `True` apparently

Reply:
Implemented it as:
```python
if (g is not None and isinstance(rv.args[0], g()) and isinstance(g()(1), TrigonometricFunction))
```
And now it works!